content/en/usage/common-problems.md

```diff
@@ -1,7 +1,258 @@
+
+## `appstraction` hangs indefinitely in `ensureDevice()` after phone crashes
+
+In large-scale analyses with `appstraction` or `cyanoacrylate`, we have sometimes [observed](https://github.com/tweaselORG/appstraction/issues/102) the phone crashing and rebooting with the following error being shown:
+
+```
+~/appstraction-test/node_modules/execa/lib/error.js:59
+		error = new Error(message);
+		        ^
+
+
+Error: Command failed with exit code 224: /home/benni/.cache/andromatic/platform-tools/adb shell am broadcast -a com.wireguard.android.action.SET_TUNNEL_UP -n 'com.wireguard.android/.model.TunnelManager$IntentReceiver' -e tunnel appstraction
+Broadcasting: Intent { act=com.wireguard.android.action.SET_TUNNEL_UP flg=0x400000 cmp=com.wireguard.android/.model.TunnelManager$IntentReceiver (has extras) }
+cmd: Failure calling service activity: Broken pipe (32)
+    at makeError (~/appstraction-test/node_modules/execa/lib/error.js:59:11)
+    at handlePromise (~/appstraction-test/node_modules/execa/index.js:124:26)
+    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
+    at Object.setProxy (~/appstraction-test/node_modules/appstraction/dist/index.js:1026:17) {
+  shortMessage: "Command failed with exit code 224: /home/benni/.cache/andromatic/platform-tools/adb shell am broadcast -a com.wireguard.android.action.SET_TUNNEL_UP -n 'com.wireguard.android/.model.TunnelManager$IntentReceiver' -e tunnel appstraction",
+  command: "/home/benni/.cache/andromatic/platform-tools/adb shell am broadcast -a com.wireguard.android.action.SET_TUNNEL_UP -n 'com.wireguard.android/.model.TunnelManager$IntentReceiver' -e tunnel appstraction",
+  escapedCommand: `"/home/benni/.cache/andromatic/platform-tools/adb" shell am broadcast -a com.wireguard.android.action.SET_TUNNEL_UP -n "'com.wireguard.android/.model.TunnelManager$IntentReceiver'" -e tunnel appstraction`,
+  exitCode: 224,
+  signal: undefined,
+  signalDescription: undefined,
+  stdout: 'Broadcasting: Intent { act=com.wireguard.android.action.SET_TUNNEL_UP flg=0x400000 cmp=com.wireguard.android/.model.TunnelManager$IntentReceiver (has extras) }\n' +
+    'cmd: Failure calling service activity: Broken pipe (32)',
+  stderr: '',
+  failed: true,
+  timedOut: false,
+  isCanceled: false,
+  killed: false
+}
+```
+
+Afterwards, starting a new analysis doesn't work because `appstraction` hangs indefinitely in `ensureDevice()`. The crash and hanging seem to be caused by Frida. To verify this, you can run:
+
+```sh
+adb shell /data/local/tmp/frida-server --version
+```
+
+If this hangs indefinitely, too, then you have the same problem. To fix the problem, you just need to kill the stuck Frida process:
+
+```sh
+adb shell pkill -9 frida-server
+```
+
+Alternatively, you can also manually reboot the phone. We are [considering](https://github.com/tweaselORG/appstraction/issues/102#issuecomment-1596111863) implementing an automatic fix for this in the future.
```

Comment:
```suggestion
Alternatively, you can also manually reboot the phone.
```
Event though we implemented "the fix", I think this is still a useful hint for troubleshooting.